### PR TITLE
JUnit attachments

### PR DESCRIPTION
**Pull request by ramilamparo**
Tasks done in this PR:
- https://github.com/Hypersequent/tms-issues/issues/329
- https://github.com/Hypersequent/tms-issues/issues/416
- https://github.com/Hypersequent/tms-issues/issues/415

## Summary by CodeRabbit

- **New Features**
  - Introduced file upload functionality via API.
  - Added new command-line script for enhanced command execution.

- **Bug Fixes**
  - Improved JSON handling in API calls.

- **Refactor**
  - Refactored JUnit upload command for better structure and functionality.
  - Introduced `JUnitCommandHandler` class for handling JUnit XML parsing and uploads.

- **Tests**
  - Added comprehensive test cases for JUnit XML uploads, including scenarios for matching and missing test cases and attachments.

- **Chores**
  - Updated dependencies and scripts in `package.json`.
  - Excluded test files from TypeScript compilation.

### CODE REVIEW

File: src/tests/junit-upload.spec.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+import { countMockedApiCalls } from './utils'
+
+const projectCode = 'TEST'
+const runId = '1'
+const domain = 'qas'
+const zone = 'eu1'
+const baseURL = `https://${domain}.${zone}.qasphere.com`
+const xmlBasePath = './src/tests/fixtures/junit-xml'
+
+const server = setupServer(
```
Comment (ramilamparo) [inline src/tests/junit-upload.spec.ts:15]: We create a mock server for unit tests.

File: src/commands/junit-upload.ts [hunk 2]
```diff
@@ -70,111 +76,27 @@ export class JUnitUploadCommandModule implements CommandModule<unknown, JUnitArg
...
-}
-
-const parseUrl = (args: Record<string, unknown>): string => {
-	if (typeof args.url === 'string') {
-		if (args.url.includes('://')) {
-			return args.url
-		}
-		return `http://${args.url}`
+		const handler = new JUnitCommandHandler(args)
+		await handler.handle()
```
Comment (ramilamparo) [inline src/commands/junit-upload.ts:100]: This is the entrypoint of the `junit-upload` command.

File: package.json [hunk 3]
```diff
@@ -43,6 +46,6 @@
 		"zod": "^3.23.7"
 	},
 	"bin": {
-		"qas-cli": "build/bin/qas-cli.js"
+		"qasphere": "build/bin/qasphere.js"
```
[the remark this answers is not in the document]
Comment (ramilamparo) [reply package.json:49]: @AndrianBdn It's not too late to rename all these into qasphere. What do you think?

**Review by AndrianBdn — APPROVED** [review]
Love this, thank you @ramilamparo 

Given that the task description was rather vague, I can't comment about something done wrong in the scope of this PR. I will try using the tool myself and set more tasks later. Code looks well structured, typed and tested. Let's run tests in GitHub Actions in the next PRs.

**Review by gornostay25 — APPROVED** [review]
Looks good! Thanks for PR @ramilamparo

**Pull request merged**